controller/TestPreviewer.php

```diff
@@ -111,4 +115,20 @@ private function getTestPreviewerPresetsMapService(): TestCategoryPresetMap
     {
         return $this->getPsrContainer()->get(TestCategoryPresetMap::class);
     }
+
+    private function mapXmlExceptionMessage(XmlStorageException $exception): string
+    {
+        if (
+            stristr(
+                $exception->getMessage(),
+                'An error occurred while unreferencing item reference with identifier'
```

Comment:
How hard is to return a specific exception instead of applying this condition???
Or maybe specify an exception code instead of checking the message...

Reply:
@andreluizmachado  seems it will require changes in QTI SDK, if so I would not go that way

Reply:
Thing is [how XmlStorageException constructor build](https://github.com/oat-sa/qti-sdk/blob/legacy/qtism/data/storage/xml/XmlStorageException.php#L50) .Its sets code always to 0 so I cant easily put there something else. I can change it but also worries me that this piece of code sits in Qti repo in branch "legacy" can we change it?